### ISSUE TITLE
Docker and demo version do not work ...

### ISSUE THREAD

**Issue by redskate** [OpenVidu/openvidu-call#32]
https://demos.openvidu.io/openvidu-call does not work

There was an error getting the token:
Http failure response for https://demos.openvidu.io:4443/api/sessions: 0 Unknown Error.

And neither the current docker version works (same error). 
In the docker version no way to "get the token" from another client.

Logging via cellular phone (firefox) gave with current verstion on my server:
```
2020-03-18 10:38:50,168 DEBG 'openvidu-server' stdout output:
[WARN] 2020-03-18 10:38:50,168 [tExec-e2-t32] io.openvidu.server.kurento.endpoint.SubscriberEndpoint (onError) - EP lsd4qxmfa2kqqhss_CAMERA_PYMVA: Failed to add ice candidate to the internal endpoint
org.kurento.jsonrpc.JsonRpcErrorException: Error adding candidate. Code: 40401. Data: {"type":"ICE_ADD_CANDIDATE_ERROR"}
	at org.kurento.jsonrpc.internal.JsonRpcRequestSenderHelper$1.onSuccess(JsonRpcRequestSenderHelper.java:128)
	at org.kurento.jsonrpc.internal.JsonRpcRequestSenderHelper$1.onSuccess(JsonRpcRequestSenderHelper.java:113)
	at org.kurento.jsonrpc.client.AbstractJsonRpcClientWebSocket$14.onSuccess(AbstractJsonRpcClientWebSocket.java:452)
	at org.kurento.jsonrpc.client.AbstractJsonRpcClientWebSocket$14.onSuccess(AbstractJsonRpcClientWebSocket.java:433)
	at com.google.common.util.concurrent.Futures$6.run(Futures.java:1773)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

and

```
2020-03-18 10:50:50,225 DEBG 'openvidu-server' stdout output:
[INFO] 2020-03-18 10:50:50,224 [tExec-e2-t64] io.openvidu.server.kurento.core.KurentoParticipantEndpointConfig (lambda$addEndpointListeners$6) - KMS event [IceComponentStateChange]: -> endpoint: hqlvepl0jiof3umt_CAMERA_YWNND (publisher) | state: FAILED | componentId: 1 | streamId: 1 | timestamp: 1584528650223
```

Basically a connection is established (the screen is divided into two sections), chat is possible, but no media (no audio and no camera) are transferred in either direction.

All ports are open. What am I doing wrong? (config is "fix" in the used docker image)

**Comment by micaelgallego** [comment]
We are experiencing a high load in our servers. We are working right now in adding more computational resources to be sure all users can use demos, particularly https://demos.openvidu.io/openvidu-call 

On the other hand, to use OpenVidu Call on your own system, you have to install OpenVidu following these instructions: 

https://openvidu.io/docs/deployment/deploying-ubuntu/

And then, install OpenVidu Call app downloading this file: 

https://github.com/OpenVidu/openvidu-call/releases/download/v2.12.0/openvidu-call-2.12.0.tar.gz

Decompress and deploying it in any web server.

To configure the OpenVidu platform used by OpenVidu Call you must include a JSON file named ov-settings.json in the root directory with the following content:
{
  "openviduCredentials": {
    "openvidu_url": "https://0.0.0.0:4443",
    "openvidu_secret": "MY_SECRET"
  }
}

You can customize other aspects of OpenVidu Call adding an extra JSON object named openviduSettings inside of the file. You have explained how openviduSettings object must be treated in detail [here](https://openvidu.io/docs/tutorials/openvidu-webcomponent.md#interface-configuration).

**Comment by dani521** [comment]
Hi I have a problem I`am extracting to web server but when I enter Room Name only white screen appears ?

**Comment by micaelgallego** [comment]
What do you mean by "I`am extracting to web server" ?

What URL are you accessing? https://demos.openvidu.io/openvidu-call/?

What problem do you have?

**Comment by dani521** [comment]
I`am trying to run on my ubuntu server followed all steps added Json File but when try to create a new session i`am receiving 401 error at token generation maybe some authentication problem ? https://openvidu.io/docs/demos/openvidu-call/ i have followed this demo

**Comment by micaelgallego** [comment]
We are working in a new way to deploy OpenVidu Call app in a server along with all necessary dependencies to make it work. It will be released during the next week.

Please stay tuned and try the the new procedure to see if it works for you

**Comment by redskate** [comment]
Hi Micael, thank you for your messages.

Do you mean the openvidu-call **docker** version is currently not to use ?
It would be very practical to have the docker version running and customizable (add volumes for SSL certificates, logs,...etc.).

Thank you 
btw: you link "here" gives a 404 ...

**Comment by micaelgallego** [comment]
The current openvidu-call docker image is not designed for production. It will be eliminated to avoid confusions.

We are working on a docker-compose version of OpenVidu Call with the ability to configure domain, change SSL certs, manage logs, configure OpenVidu, etc.

This is currently in development, but you can take a try and provide feedback to us: https://github.com/OpenVidu/openvidu/blob/deploy-docker-compose/openvidu-server/docker/openvidu-docker-compose/readme.md

Please note that instructions are incomplete because the code is still in a branch. You will have to change to that branch after cloning the repository. 

Also, take into account that this is alpha software. We are polishing things and improving documentation.

**Comment by NiceboyWiseboy** [comment]
> Hi I have a problem I`am extracting to web server but when I enter Room Name only white screen appears ?

**Comment by NiceboyWiseboy** [comment]
I'm also having the same issue which @dani521 faced earlier.

**Comment by NiceboyWiseboy** [comment]
I'm putting some Screenshots for your better understanding:

![screenshot1](https://user-images.githubusercontent.com/40816700/79344612-af640b00-7f4d-11ea-803b-55007bb0b209.png)

![screenshot2](https://user-images.githubusercontent.com/40816700/79344610-ae32de00-7f4d-11ea-8908-66dfacfe27ce.png)

**Comment by CSantosM** [comment]
Hello , 

Which docker are you using? 
Could you see if shows any error in the console?

**Comment by micaelgallego** [comment]
You have to use https to connect to a WebRTC application.

**Comment by NiceboyWiseboy** [comment]
Hello @CSantosM ,

I'm not using OpenViduCall by docker, but using [this ](http://docs.openvidu.io/en/2.12.0/demos/openvidu-call/#using-the-release)guide. This guide shows to install the OpenViduCall from Releases page.

**Comment by NiceboyWiseboy** [comment]
Hello @micaelgallego ,

I followed this [guide ](http://docs.openvidu.io/en/2.12.0/demos/openvidu-call/#using-the-release)from OpenVidu docs which says I don't SSL for OpenViduCall. It says I only needed Self-Signed certificate accepted from the OpenVidu server which I had installed before OpenViduCall. I done the same but it only shows the WHITE SCREEN which I posted earlier.

I'll try the new solution which you've provided me with.

**Comment by micaelgallego** [comment]
As specified in that guide, your app works on `http://localhost:8080`, but if you try to access to the web from other domain or IP different than localhost, you will need https.

**Comment by NiceboyWiseboy** [comment]
![image](https://user-images.githubusercontent.com/40816700/79412649-a0b83b00-7fc3-11ea-8826-a52f3be5be21.png)


This is what I get if I call the URL with **https**. So, I have to make SSL certificate and then access the application, then it'll work, right?

**Comment by micaelgallego** [comment]
I think so.

Can see share with us the console log of the browser? To see what kind of errors you are having.
